################################################################################
SHORT DESCRIPTION:
################################################################################

################################################################################
LONG DESCRIPTION:
################################################################################
This patch contains several changes that couldn't be grouped elsewhere. These
changes support running xl on OpenXT with xenmgr as the toplevel toolstack
component. Some of the changes include tweaking qemu options, and writing
OpenXT specific xenstore nodes.  Refer to the patch below for a comprehensive
list of changes.

################################################################################
CHANGELOG
################################################################################
Authors:
Chris Rogers <rogersc@ainfosec.com>

Updated during Xen 4.8, 4.9 uprevs.

Disaggregated into multiple patches by:
Troy Crosley <crosleyt@ainfosec.com>

################################################################################
REMOVAL
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################

################################################################################
INTERNAL DEPENDENCIES
################################################################################

################################################################################
PATCHES
################################################################################
--- a/tools/libxl/libxl_domain.c
+++ b/tools/libxl/libxl_domain.c
@@ -1824,7 +1824,7 @@ int libxl_retrieve_domain_configuration(
                         break;
                 }
 
-                if (j < num) {         /* found in xenstore */
+                if ((j < num) || dt->type == LIBXL__DEVICE_KIND_VIF) { /* found in xenstore */
                     if (dt->merge)
                         dt->merge(ctx, p + dt->dev_elem_size * j, q);
                 } else {                /* not found in xenstore */
--- a/tools/libxl/libxl_nic.c
+++ b/tools/libxl/libxl_nic.c
@@ -129,6 +129,8 @@ static int libxl__set_xenstore_nic(libxl
                                    flexarray_t *back, flexarray_t *front,
                                    flexarray_t *ro_front)
 {
+    libxl_uuid uuid;
+
     flexarray_grow(back, 2);
 
     if (nic->script)
@@ -236,6 +238,9 @@ static int libxl__set_xenstore_nic(libxl
     flexarray_append(front, "mac");
     flexarray_append(front, GCSPRINTF(
                                     LIBXL_MAC_FMT, LIBXL_MAC_BYTES(nic->mac)));
+    libxl_domid_to_uuid(CTX, &uuid, nic->backend_domid);
+    flexarray_append(front, "backend-uuid");
+    flexarray_append(front, GCSPRINTF("%s", libxl__uuid2string(gc, uuid)));
 
     return 0;
 }
